sync/patches/chart.yaml/patch.sh

```diff
@@ -0,0 +1,23 @@
+#!/usr/bin/env bash
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+repo_dir=$(git rev-parse --show-toplevel) ; readonly repo_dir
+#script_dir=$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd ) ; readonly script_dir
+
+cd "${repo_dir}"
+
+#readonly script_dir_rel=".${script_dir#"${repo_dir}"}"
+
+set -x
+sed -i \
+        -e '/^version:/c\version: 0.25.1' \
```

Comment:
one problem here is that this version is not static. it will change on every release, but has to be updated here also.

Reply:
I'm torn here.
The actual value of `version` field in `Chart.yaml` is being changed by app-build-suite in CI for the bundled chart archive. The release automation also changes the `version` to prevent people needing to manually bump the field and people being confused about the difference between released and comitted `Chart.yaml` files.

Devs will need to fix the line in question by hand now, because the release PR will fail because CI will update the value in `Chart.yaml`.

Reply:
here https://github.com/giantswarm/envoy-gateway-app/pull/7/files#diff-d6bc74a267692780025826cf61ef5d9af36719448b763e25b43ce111bc5fdf05R18 I actually gave up and ignored the file completely.

Reply:
Thinking about it, I find myself liking your solution. This would also fit kong, and we can keep relying on the release automation that exists. I'll try to adapt this

Reply:
I don't understand why this is better than changing it in the patch? It still has to be updated manually.

Reply:
> It still has to be updated manually

no, once the "gs" version is there, our automation will take care.

Reply:
Should I just set `0.0.0-dev` here?

Reply:
hmm if you want to patch this file every time we sync with upstream, the version should be the latest of the repo. I think that's too much for a file that should only contain what we need it to contain. There is not real reason to sync it with upstream in regular basis.
That's why I suggest just to add it to the ignorePath and craft our own.

Reply:
Ok. Diff will show the major differences anyway.

Reply:
All of that will be changing. I personally prefer to have a patch here. It's less manual work.

```
appVersion: 1.15.6
kubeVersion: ">= 1.16.0-0"
icon: https://cdn.jsdelivr.net/gh/cilium/cilium@v1.15/Documentation/images/logo-solo.svg
```

Reply:
but what would you do with `version`? our release process changes the value every time we do a release. if that value is not reflected somehow here it will get reverted to the cilium version on each sync.sh

Reply:
Lets try to make a release by creating a branch `replace-upstream-with-patches#release#minor` from this change (setting `0.0.0-dev`). I think the newly introduced CI check will fail because the release PR automation changes the `helm/**/Chart.yaml` `version` field.